### PR TITLE
Fix part #646: Low-fi Introduce Subtopic in Practice Mode

### PR DESCRIPTION
**Pull request by veena14cs**
## Explanation
This PR focuses on introducing subtopic in app layer module that is used in Practice Tab.

### Design Doc
https://docs.google.com/document/d/1ZH3HWkQeVipkOGQLBTjp6Rjq7z_ac05jHbDXYQq0EN4/edit?usp=sharing

### Mock 
https://xd.adobe.com/view/0687f00c-695b-437a-79a6-688e7f4f7552-70b6/screen/f403add1-079a-4202-9975-efe04c076290/TP-Train-Unchecked-

## Checklist

- [ ] The PR title starts with "Fix #bugnum: ", followed by a short, clear summary of the changes. (If this PR fixes part of an issue, prefix the title with "Fix part of #bugnum: ...".)
- [ ] The PR explanation includes the words "Fixes #bugnum: ..." (or "Fixes part of #bugnum" if the PR only partially fixes an issue).
- [x] The PR follows the [style guide](https://github.com/oppia/oppia-android/wiki/Coding-style-guide).
- [x] The PR does not contain any unnecessary auto-generated code from Android Studio.
- [x] The PR is made from a branch that's **not** called "develop".
- [x] The PR is made from a branch that is up-to-date with "develop".
- [ ] The PR's branch is based on "develop" and not on any other branch.
- [x] The PR is **assigned** to an appropriate reviewer in both the **Assignees** and the **Reviewers** sections.

### CODE REVIEW

File: app/src/main/java/org/oppia/app/topic/practice/SkillSelector.kt [hunk 1]
```diff
@@ -2,9 +2,9 @@ package org.oppia.app.topic.practice
 
 /** Interface to update the selectedSkillList in [TopicPracticeFragmentPresenter]. */
```
Comment (rt4914) [inline app/src/main/java/org/oppia/app/topic/practice/SkillSelector.kt:3]: Update this comment also.

File: app/src/main/java/org/oppia/app/topic/practice/SkillSelector.kt [hunk 2]
```diff
@@ -2,9 +2,9 @@ package org.oppia.app.topic.practice
...
 interface SkillSelector {
```
Comment (rt4914) [inline app/src/main/java/org/oppia/app/topic/practice/SkillSelector.kt]: Shouldn't this be `SubtopicSelector` ?

File: app/src/main/java/org/oppia/app/topic/practice/TopicPracticeFragmentPresenter.kt [hunk 3]
```diff
@@ -111,7 +114,9 @@ class TopicPracticeFragmentPresenter @Inject constructor(
     binding.viewModel = model
     binding.isSubmitButtonActive = selectedSkillIdList.isNotEmpty()
     binding.topicPracticeStartButton.setOnClickListener {
-      routeToQuestionPlayerListener.routeToQuestionPlayer(selectedSkillIdList)
+      val skillIdList = ArrayList(skillIdHashmap.values)
+      logger.d("TopicPracticeFragmentPresenter","Skill IDs = "+skillIdList.flatten())
```
Comment (rt4914) [inline app/src/main/java/org/oppia/app/topic/practice/TopicPracticeFragmentPresenter.kt]: Nit:
`logger.d("TopicPracticeFragmentPresenter", "Skill Ids = " + skillIdList.flatten())`

File: app/src/main/java/org/oppia/app/topic/practice/TopicPracticeFragmentPresenter.kt [hunk 4]
```diff
@@ -17,20 +17,23 @@ import org.oppia.app.topic.TOPIC_ID_ARGUMENT_KEY
...
 class TopicPracticeFragmentPresenter @Inject constructor(
   private val activity: AppCompatActivity,
   private val fragment: Fragment,
+  private val logger: Logger,
   private val viewModelProvider: ViewModelProvider<TopicPracticeViewModel>
 ) : SkillSelector {
   private lateinit var binding: TopicPracticeFragmentBinding
   private lateinit var linearLayoutManager: LinearLayoutManager
   lateinit var selectedSkillIdList: ArrayList<String>
+  private  var skillIdHashmap = HashMap<String,MutableList<String>>()
```
Comment (rt4914) [inline app/src/main/java/org/oppia/app/topic/practice/TopicPracticeFragmentPresenter.kt]: rename: `skillIdHashMap`

File: app/src/main/java/org/oppia/app/topic/practice/practiceitemviewmodel/TopicPracticeSubtopicViewModel.kt [hunk 5]
```diff
@@ -1,7 +1,7 @@
 package org.oppia.app.topic.practice.practiceitemviewmodel
 
-import org.oppia.app.model.SkillSummary
+import org.oppia.app.model.Subtopic
 import org.oppia.app.topic.practice.TopicPracticeFragment
 
 /** Skill summary view model for the recycler view in [TopicPracticeFragment]. */
```
Comment (rt4914) [inline app/src/main/java/org/oppia/app/topic/practice/practiceitemviewmodel/TopicPracticeSubtopicViewModel.kt]: Update comment

File: app/src/main/res/layout/topic_practice_skill_view.xml [hunk 6]
```diff
@@ -10,7 +10,7 @@
 
     <variable
```
Comment (rt4914) [inline app/src/main/res/layout/topic_practice_skill_view.xml:11]: Rename file to `topic_practice_subtopic`

**Review by rt4914 — CHANGES_REQUESTED** [review]
Nit changes suggested.

File: app/src/main/java/org/oppia/app/topic/practice/SkillSelector.kt (hunk 1 above)
Comment (veena14cs) [reply app/src/main/java/org/oppia/app/topic/practice/SkillSelector.kt:3]: Done.

File: app/src/main/java/org/oppia/app/topic/practice/SkillSelector.kt (hunk 2 above)
Comment (veena14cs) [reply app/src/main/java/org/oppia/app/topic/practice/SkillSelector.kt]: Done,

File: app/src/main/java/org/oppia/app/topic/practice/TopicPracticeFragmentPresenter.kt (hunk 3 above)
Comment (veena14cs) [reply app/src/main/java/org/oppia/app/topic/practice/TopicPracticeFragmentPresenter.kt]: Done.

File: app/src/main/java/org/oppia/app/topic/practice/TopicPracticeFragmentPresenter.kt (hunk 4 above)
Comment (veena14cs) [reply app/src/main/java/org/oppia/app/topic/practice/TopicPracticeFragmentPresenter.kt]: Done.

**Review by nikitamarysolomanpvt — APPROVED** [review]
LGTM. Please address comments of rajat.

File: app/src/main/java/org/oppia/app/topic/practice/practiceitemviewmodel/TopicPracticeSubtopicViewModel.kt (hunk 5 above)
Comment (veena14cs) [reply app/src/main/java/org/oppia/app/topic/practice/practiceitemviewmodel/TopicPracticeSubtopicViewModel.kt]: Done.

File: app/src/main/res/layout/topic_practice_skill_view.xml (hunk 6 above)
Comment (veena14cs) [reply app/src/main/res/layout/topic_practice_skill_view.xml:11]: Done.

**Review by rt4914 — APPROVED** [review]
LGTM, thanks.

**Pull request merged**